src/mono/wasm/runtime/loader/polyfills.ts

```diff
@@ -54,13 +54,12 @@ export async function detect_features_and_polyfill(module: DotnetModuleInternal)
         if (isPathAbsolute(path)) return path;
         return loaderHelpers.scriptDirectory + path;
     };
-    loaderHelpers.downloadResource = module.downloadResource;
     loaderHelpers.fetch_like = fetch_like;
     // eslint-disable-next-line no-console
     loaderHelpers.out = console.log;
     // eslint-disable-next-line no-console
     loaderHelpers.err = console.error;
-    loaderHelpers.getApplicationEnvironment = module.getApplicationEnvironment;
+    loaderHelpers.onDownloadResourceProgress = module.onDownloadResourceProgress;
```

Comment:
Maybe we could stop adding APIs to the `module` object and do `withOnDownloadResourceProgress` in run.ts

Reply:
This one is already there and blazor uses it, but otherwise I agree

Reply:
Let's drop it form `DotnetModuleConfig` and add the API they could adopt going forward.

Reply:
In follow up, as discussed offline